Rename EraSegWits to EraBlockBody and TxSeq to BlockBody

aniketd opened the pull request:
# Description

Closes #5121 

# Checklist

- [x] Commits in meaningful sequence and with useful messages.
- [ ] ~~Tests added or updated when needed.~~
- [x] `CHANGELOG.md` files updated for packages with externally visible changes.  
      **NOTE: _New section is never added with the code changes._** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#changelogmd)).
- [ ] ~~Versions updated in `.cabal` and `CHANGELOG.md` files when necessary, according to the
      [versioning process](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process).~~
- [ ] ~~Version bounds in `.cabal` files updated when necessary.  
      **NOTE: _If bounds change in a cabal file, that package itself must have a version increase._** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process)).~~
- [x] Code formatted (use `scripts/fourmolize.sh`).
- [x] Cabal files formatted (use `scripts/cabal-format.sh`).
- [x] CDDL files are up to date (use `scripts/gen-cddl.sh`)
- [x] [`hie.yaml`](https://github.com/intersectmbo/cardano-ledger/blob/master/hie.yaml) updated (use `scripts/gen-hie.sh`).
- [x] Self-reviewed the diff.

lehins commented on libs/cardano-ledger-core/src/Cardano/Ledger/Core.hs:
```diff
@@ -554,51 +555,66 @@ hashScript =
-  hashTxSeq :: TxSeq era -> Hash.Hash HASH EraIndependentBlockBody
+  hashBlockBody :: BlockBody era -> Hash.Hash HASH EraIndependentBlockBody
+
+  hashTxSeq :: BlockBody era -> Hash.Hash HASH EraIndependentBlockBody
```
Since `hashBlockBody` is the replacement we can use default implementation and remove custom implementation from all eras. This will simplify its later removal.
```suggestion
  hashTxSeq :: BlockBody era -> Hash.Hash HASH EraIndependentBlockBody
  hashTxSeq = hashBlockBody
```

lehins commented on libs/cardano-ledger-core/src/Cardano/Ledger/Core.hs:
```diff
@@ -554,51 +555,66 @@ hashScript =
-  toTxSeq :: StrictSeq (Tx era) -> TxSeq era
+  mkBasicBlockBody :: BlockBody era
+  default mkBasicBlockBody :: BlockBody era
+  mkBasicBlockBody = toTxSeq mempty
```
`toTxSeq` has been deprecated, so you shouldn't be using it for default implementation
```suggestion
```

Also `default` with `DefaultSignaures` is only needed when type signature is different, which was not the case here.

lehins commented on libs/cardano-protocol-tpraos/testlib/Test/Cardano/Protocol/TPraos/Arbitrary.hs:
```diff
@@ -149,26 +150,32 @@ deriving newtype instance Arbitrary KESPeriod
+genBlock aiks =
+  Block
+    <$> genBHeader aiks
+    <*> ((\x -> mkBasicBlockBody & txSeqBlockBodyL .~ x) <$> arbitrary)
```
Although this is correct today, this will not necessarily be correct tomorrow. Imagine block body becomes more complicated and the type contains something else, besides the sequence. This implementation will never generate it.
```suggestion
    <*> arbitrary
```
You'll need to add `Arbitrary (BlockBody era)` constraint in order for this to compile

lehins commented on eras/shelley/impl/src/Cardano/Ledger/Shelley/BlockBody/Internal.hs:
```diff
@@ -69,48 +75,50 @@ import Data.Sequence.Strict (StrictSeq)
-  { txSeqTxns' :: !(StrictSeq (ShelleyTx era))
-  , txSeqHash :: Hash.Hash HASH EraIndependentBlockBody
+data ShelleyBlockBody era = BlockBody'
+  { bbTxns :: !(StrictSeq (ShelleyTx era))
```
Since you are already renaming them, let's give those fields better names. Also you forgot `s` suffix for Shelley
```suggestion
  { sbbTxs :: !(StrictSeq (ShelleyTx era))
```

lehins commented on eras/shelley/impl/src/Cardano/Ledger/Shelley/BlockBody/Internal.hs:
```diff
@@ -69,48 +75,50 @@ import Data.Sequence.Strict (StrictSeq)
+  , bbHash :: Hash.Hash HASH EraIndependentBlockBody
   -- ^ Memoized hash to avoid recomputation. Lazy on purpose.
-  , txSeqBodyBytes :: BSL.ByteString
+  , bbBodyBytes :: BSL.ByteString
```
```suggestion
  , sbbTxsBodyBytes :: BSL.ByteString
```

lehins commented on eras/shelley/impl/src/Cardano/Ledger/Shelley/BlockBody/Internal.hs:
```diff
@@ -69,48 +75,50 @@ import Data.Sequence.Strict (StrictSeq)
+  , bbBodyBytes :: BSL.ByteString
   -- ^ Bytes encoding @Seq ('TxBody' era)@
-  , txSeqWitsBytes :: BSL.ByteString
+  , bbWitsBytes :: BSL.ByteString
```
```suggestion
  , sbbTxsWitsBytes :: BSL.ByteString
```

lehins commented on eras/shelley/impl/src/Cardano/Ledger/Shelley/BlockBody/Internal.hs:
```diff
@@ -69,48 +75,50 @@ import Data.Sequence.Strict (StrictSeq)
+  , bbWitsBytes :: BSL.ByteString
   -- ^ Bytes encoding @Seq ('TxWits' era)@
-  , txSeqMetadataBytes :: BSL.ByteString
+  , bbMetadataBytes :: BSL.ByteString
```
```suggestion
  , sbbTxsAuxDataBytes :: BSL.ByteString
```

lehins commented on eras/shelley/impl/src/Cardano/Ledger/Shelley/BlockBody/Internal.hs:
```diff
@@ -16,11 +16,17 @@
+-- The contents of this module may change __in any way whatsoever__
+-- and __without any warning__ between minor versions of this package.
+module Cardano.Ledger.Shelley.BlockBody.Internal (
+  ShelleyBlockBody (ShelleyBlockBody, bbHash, BlockBody'),
```
Because this module is internal, it is now ok to export all fields
```suggestion
  ShelleyBlockBody (ShelleyBlockBody, ..),
```

lehins commented on eras/shelley/impl/src/Cardano/Ledger/Shelley/BlockBody/Internal.hs:
```diff
@@ -16,11 +16,17 @@
 {-# LANGUAGE UndecidableInstances #-}
 {-# OPTIONS_GHC -Wno-orphans #-}
 
```
We need to indicate to `haddock` to not generate any pages for this module
```suggestion
{-# OPTIONS_HADDOCK not-home #-}
```

lehins commented on eras/shelley/impl/src/Cardano/Ledger/Shelley/BlockBody/Internal.hs:
```diff
@@ -69,48 +75,50 @@ import Data.Sequence.Strict (StrictSeq)
-data ShelleyTxSeq era = TxSeq'
-  { txSeqTxns' :: !(StrictSeq (ShelleyTx era))
-  , txSeqHash :: Hash.Hash HASH EraIndependentBlockBody
+data ShelleyBlockBody era = BlockBody'
```
Let's name it something more sensible.
```suggestion
data ShelleyBlockBody era = ShelleyBlockBodyInternal
```
Note that I did not suggest `Raw` suffix in order to disambiguate from `MemoBytes`, where Raw types do not have bytes associated with them, which is not the case here.

lehins commented on eras/shelley/impl/src/Cardano/Ledger/Shelley/BlockBody.hs:
```diff
@@ -0,0 +1,11 @@
+module Cardano.Ledger.Shelley.BlockBody (
+  ShelleyBlockBody (ShelleyBlockBody, bbHash),
```
It is unsafe to export `bbHash` from public API, since it also acts as a modifier. 

I'd suggest replacing all it's uses with `hashBlockBody` instead.

```suggestion
  ShelleyBlockBody (ShelleyBlockBody),
```

Alternatively, you could add a separate function `blockBodyHash` and export that function instead:

```haskell
blockBodyHash :: Hash.Hash HASH EraIndependentBlockBody
blockBodyHash = bbHash
```

```suggestion
  ShelleyBlockBody (ShelleyBlockBody),
  blockBodyHash,
```

lehins commented on eras/alonzo/impl/src/Cardano/Ledger/Alonzo/BlockBody/Internal.hs:
```diff
@@ -64,48 +63,50 @@ import NoThunks.Class (AllowThunksIn (..), NoThunks)
-data AlonzoTxSeq era = AlonzoTxSeqRaw
-  { txSeqTxns :: !(StrictSeq (Tx era))
-  , txSeqHash :: Hash.Hash HASH EraIndependentBlockBody
+data AlonzoBlockBody era = AlonzoBlockBodyRaw
```
Let's call it "internal" instead, to not confuse with raw types used by MemoBytes
```suggestion
data AlonzoBlockBody era = AlonzoBlockBodyInternal
```

lehins commented on eras/alonzo/impl/src/Cardano/Ledger/Alonzo/BlockBody/Internal.hs:
```diff
@@ -64,48 +63,50 @@ import NoThunks.Class (AllowThunksIn (..), NoThunks)
+  , abbHash :: Hash.Hash HASH EraIndependentBlockBody
   -- ^ Memoized hash to avoid recomputation. Lazy on purpose.
-  , txSeqBodyBytes :: BSL.ByteString
+  , abbBodyBytes :: BSL.ByteString
```
```suggestion
  , abbTxsBodyBytes :: BSL.ByteString
```

lehins commented on eras/alonzo/impl/src/Cardano/Ledger/Alonzo/BlockBody/Internal.hs:
```diff
@@ -64,48 +63,50 @@ import NoThunks.Class (AllowThunksIn (..), NoThunks)
+  , abbBodyBytes :: BSL.ByteString
   -- ^ Bytes encoding @Seq ('TxBody' era)@
-  , txSeqWitsBytes :: BSL.ByteString
+  , abbWitsBytes :: BSL.ByteString
```
```suggestion
  , abbTxsWitsBytes :: BSL.ByteString
```

lehins commented on eras/alonzo/impl/src/Cardano/Ledger/Alonzo/BlockBody/Internal.hs:
```diff
@@ -64,48 +63,50 @@ import NoThunks.Class (AllowThunksIn (..), NoThunks)
+  , abbWitsBytes :: BSL.ByteString
   -- ^ Bytes encoding @Seq ('TxWits' era)@
-  , txSeqMetadataBytes :: BSL.ByteString
+  , abbMetadataBytes :: BSL.ByteString
```
```suggestion
  , abbTxsAuxDataBytes :: BSL.ByteString
```

lehins commented on eras/alonzo/impl/src/Cardano/Ledger/Alonzo/BlockBody/Internal.hs:
```diff
@@ -64,48 +63,50 @@ import NoThunks.Class (AllowThunksIn (..), NoThunks)
   -- ^ Bytes encoding a @'TxAuxData')@. Missing indices have
   -- 'SNothing' for metadata
-  , txSeqIsValidBytes :: BSL.ByteString
+  , abbIsValidBytes :: BSL.ByteString
```
```suggestion
  , abbTxsIsValidBytes :: BSL.ByteString
```

lehins commented on eras/alonzo/impl/src/Cardano/Ledger/Alonzo/BlockBody/Internal.hs:
```diff
@@ -64,48 +63,50 @@ import NoThunks.Class (AllowThunksIn (..), NoThunks)
-  { txSeqTxns :: !(StrictSeq (Tx era))
-  , txSeqHash :: Hash.Hash HASH EraIndependentBlockBody
+data AlonzoBlockBody era = AlonzoBlockBodyRaw
+  { abbTxns :: !(StrictSeq (Tx era))
```
```suggestion
  { abbTxs :: !(StrictSeq (Tx era))
```

lehins reviewed (changes requested):
Few minor things can be improved, but other than that it looks good.
I'd also like to suggest adding a separate commit that brings back old modules, marks those modules as deprecated and reexports as many of the old functions as possible, plus some compatibility shims, eg:
```haskell
module Cardano.Ledger.Alonzo.TxSeq
  {-# DEPRECATED "Use `Cardano.Ledger.Alonzo.BlockBody` instead" #-} (
  type AlonzoTxSeq,
  pattern AlonzoTxSeq,
  txSeqTxns,
) where

import Cardano.Ledger.Alonzo.BlockBody.Internal

type AlonzoTxSeq = AlonzoBlockBody

pattern AlonzoTxSeq = AlonzoBlockBody

txSeqTxns = abbTxs
```

lehins commented on eras/alonzo/test-suite/test/Test/Cardano/Ledger/Alonzo/Golden.hs:
```diff
@@ -296,13 +293,13 @@ goldenMinFee =
               case decodeFullAnnotator (eraProtVerHigh @AlonzoEra) "Block" decCBOR cborBytesBlock of
                 Left err -> error (show err)
-                Right (Block _h txs :: Block (BHeader StandardCrypto) AlonzoEra) -> txs
+                Right blockBody' -> blockBody'
```
This is likely why the test is failing on CI. You confused `Block` with `BlockBody`. `Block` also has a block header, which is encoded in CBOR, but you changed it in such a way that it now tries to decode BlockBody instead of a Block
```suggestion
                Right block -> bbody block
```

lehins commented on eras/babbage/impl/src/Cardano/Ledger/Babbage/Tx.hs:
```diff
@@ -47,9 +44,9 @@ instance AlonzoEraTx BabbageEra where
+  type BlockBody BabbageEra = AlonzoBlockBody BabbageEra
+  mkBasicBlockBody = mkBasicBlockBody
+  txSeqBlockBodyL = txSeqBlockBodyL
+  hashBlockBody = hashBlockBody
```
What is that?
The only reason why this type checks is because this implementation results in bottom.
It's the same as if you wrote:
```haskell
instance Show Foo where
  show = show
```
It will type check but if you call `show` on `Foo` it will never terminate.

This is the reason why some test suites never terminate and are killed with a timeout.

aniketd replied:
What a blooper! Thanks! :grin:

lehins commented on eras/allegra/impl/src/Cardano/Ledger/Allegra/BlockBody.hs:
```diff
@@ -3,21 +3,20 @@
+  type BlockBody AllegraEra = ShelleyBlockBody AllegraEra
+  mkBasicBlockBody = mkBasicBlockBody
+  txSeqBlockBodyL = txSeqBlockBodyL
+  hashBlockBody = hashBlockBody
```
Ditto. Needs fixing

lehins commented on eras/conway/impl/src/Cardano/Ledger/Conway/Tx.hs:
```diff
@@ -121,9 +118,9 @@ instance AlonzoEraTx ConwayEra where
+  type BlockBody ConwayEra = AlonzoBlockBody ConwayEra
+  mkBasicBlockBody = mkBasicBlockBody
+  txSeqBlockBodyL = txSeqBlockBodyL
+  hashBlockBody = hashBlockBody
```
Ditto

lehins commented on eras/dijkstra/src/Cardano/Ledger/Dijkstra/Tx.hs:
```diff
@@ -52,9 +52,9 @@ instance AlonzoEraTx DijkstraEra where
+  type BlockBody DijkstraEra = AlonzoBlockBody DijkstraEra
+  mkBasicBlockBody = mkBasicBlockBody
+  txSeqBlockBodyL = txSeqBlockBodyL
+  hashBlockBody = hashBlockBody
```
Ditto

lehins commented on eras/mary/impl/src/Cardano/Ledger/Mary/BlockBody.hs:
```diff
@@ -3,21 +3,20 @@
+  type BlockBody MaryEra = ShelleyBlockBody MaryEra
+  mkBasicBlockBody = mkBasicBlockBody
+  txSeqBlockBodyL = txSeqBlockBodyL
+  hashBlockBody = hashBlockBody
```
Ditto

lehins commented on eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Rules/TestChain.hs:
```diff
@@ -253,13 +253,13 @@ ledgerTraceBase chainSt block =
     pp_ = nes ^. curPParamsEpochStateL
     -- Oldest to Newest first
-    txs = (reverse . toList . fromTxSeq) txSeq -- HERE WE USE SOME SegWit function
+    txs = blockBody ^. txSeqBlockBodyL . to toList . to reverse -- HERE WE USE SOME SegWit function
```
I would like to suggest avoiding using `to` like this in general, unless you need to chain more lenses afterwards, eg `txBody ^. inputsTxBodyL . to toList . to head . coinTxOutL`. In other words there is no need to lensify everything.
```suggestion
    txs = reverse $ toList $ blockBody ^. txSeqBlockBodyL -- HERE WE USE SOME SegWit function
```

lehins commented on eras/shelley/test-suite/test/Test/Cardano/Ledger/Shelley/Serialisation/Golden/Encoding.hs:
```diff
@@ -333,7 +332,7 @@ testBHB =
           (vrfSignKey testVRF)
     , bsize = 0
-    , bhash = bbHash @era $ ShelleyTxSeq @era StrictSeq.empty
+    , bhash = hashBlockBody $ mkBasicBlockBody & txSeqBlockBodyL @era .~ StrictSeq.empty
```
No need to overwrite empty with empty
```suggestion
    , bhash = hashBlockBody $ mkBasicBlockBody @era
```

lehins commented on eras/shelley/test-suite/test/Test/Cardano/Ledger/Shelley/Serialisation/Golden/Encoding.hs:
```diff
@@ -900,7 +898,7 @@ tests =
           size = 0
           blockNo = BlockNo 44
-          bbhash = bbHash @C $ ShelleyTxSeq StrictSeq.empty
+          bbhash = hashBlockBody @C $ ShelleyBlockBody StrictSeq.empty
```
```suggestion
          bbhash = hashBlockBody $ mkBasicBlockBody @C
```

lehins commented on eras/shelley/test-suite/test/Test/Cardano/Ledger/Shelley/Serialisation/Golden/Encoding.hs:
```diff
@@ -978,7 +976,7 @@ tests =
           sig = unsoundPureSignedKES () 0 (testBHB @C) (kesSignKey testKESKeys)
           bh = BHeader (testBHB @C) sig
-          txns = ShelleyTxSeq StrictSeq.Empty
+          txns = ShelleyBlockBody StrictSeq.Empty
```
```suggestion
          txns = mkBasicBlockBody
```

lehins commented on libs/cardano-ledger-core/src/Cardano/Ledger/Block.hs:
```diff
@@ -113,12 +113,12 @@ bbody (Block _ txs) = txs
+neededTxInsForBlock (Block _ blockBody) = Set.filter isNotNewInput allTxIns
   where
-    txBodies = map (^. bodyTxL) $ toList $ fromTxSeq txsSeq
+    txBodies = map (^. bodyTxL) $ blockBody ^. txSeqBlockBodyL . to toList
```
```suggestion
    txBodies = map (^. bodyTxL) $ toList $ blockBody ^. txSeqBlockBodyL
```

lehins commented on libs/cardano-protocol-tpraos/testlib/Test/Cardano/Protocol/TPraos/Arbitrary.hs:
```diff
@@ -149,26 +150,31 @@ deriving newtype instance Arbitrary KESPeriod
+  arbitrary =
+    Block
+      <$> arbitrary
+      <*> ((\x -> mkBasicBlockBody & txSeqBlockBodyL .~ x) <$> arbitrary)
```
Same comment as I had for `genBlock`
```suggestion
      <*> arbitrary
```

lehins replied:
It doesn't look like this comment was addressed

aniketd replied:
This was my mistake during the rebase. I was fixing it during self-review yesterday :stuck_out_tongue: and you posted your review a little while after :grin:

lehins reviewed (changes requested):
All of the same problems are still there?!
Did you forget to push to remote?
Also needs a rebase.

aniketd committed:
Rename EraSegWits->EraBlockBody, TxSeq->BlockBody.

- Rename EraSegWits and TxSeq.
  - class `EraSegWits` to `EraBlockBody`, and
  - type `TxSeq` to `BlockBody`.
- Add mkBasicBlockBody to EraBlockBody
- Add txSeqBlockBodyL lens to EraBlockBody
- Deprecate fromTxSeq and toTxSeq in favour of txSeqBlockBodyL
- Add hashBlockBody to EraBlockBody and deprecate hashTxSeq in favour of
  it.
- Keep it exposed, since testlib depends on the TxSeq' constructor in a
  DecCBOR instance.
- Move Shelley.BlockChain to an Internal module.
- Rename Shelley.BlockChain to Shelley.BlockBody
- Rename Alonzo.TxSeq to Alonzo.BlockBody and the corresponding Internal
  module too.
- Rename ShelleyTxSeq to ShelleyBlockBody the data-type as well as the
  pattern.
- Replace the superfluous accessor function `bbHash` with the actual
  field accessor, and export it directly.
- Add missing Arbitrary constraints for BlockBody types.
- Restore deleted modules with DEPRECATED warnings.
  - also bring back the functions, types and constructors so that
    downstream is notified by warnings rather than errors.
- Don't export hash accessor, use hashBlockBody instead
- Rename internal blockbody data constructors
- Express deprecated EraBlockBody members in terms of their new
  counterparts
- Rename {Allegra,Mary}.TxSeq to BlockBody
- Rename ShelleyBlockBody constructor and fields.
  - TxSeq' to ShelleyBlockBody
  - txSeqTxns' to sbbTxs
  - txSeqHash to sbbHash
  - txSeqBodyBytes to sbbTxsBodyBytes
  - txSeqWitsBytes to sbbTxsWitsBytes
  - txSeqMetadataBytes to sbbTxsAuxDataBytes
- Replace the superfluous accessor function `hashAlonzoTxSeq` with the
  actual field accessor, and export it directly.
- Rename AlonzoBlockBody constructor and fields.
  - AlonzoTxSeqRaw to AlonzoBlockBodyRaw
  - txSeqTxns to abbTxs
  - txSeqHash to abbHash
  - txSeqBodyBytes to abbTxsBodyBytes
  - txSeqWitsBytes to abbTxsWitsBytes
  - txSeqMetadataBytes to abbTxsAuxDataBytes
  - txSeqIsValidBytes to abbTxsIsValidBytes
- Move EraBlockBody instances for Babbage and Conway into their own
  module.

aniketd commented:
I forgot to actually push. I have rebased again over even more changes from master now.

aniketd commented on eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/Rules/AdaPreservation.hs:
```diff
@@ -269,7 +269,7 @@ checkPreservation SourceSignalTarget {source, target, signal} count =
             ]
 
-    txs' = toList $ (fromTxSeq @era . bbody) signal
+    txs' = signal ^. to bbody . txSeqBlockBodyL . to toList
```
TODO @aniketd

lehins commented on eras/mary/impl/src/Cardano/Ledger/Mary/BlockBody.hs:
```diff
@@ -0,0 +1,27 @@
+  ShelleyBlockBody (ShelleyBlockBody),
+  sbbHash,
+  sbbTxs,
+ )
```
Instead of importing an `Internal` module, you need to create and export safe versions of these functions.
```suggestion
import Cardano.Ledger.Shelley.BlockBody (
  ShelleyBlockBody (ShelleyBlockBody),
  shelleyBlockBodyHash,
  shelleyBlockBodyTxs,
 )
```

This is a general rule of thumb. If we don't have to rely on the internals, we should do our best to use safe functionality. Not only it promotes safer code, but it also serves as an example for any downstream user that is reading our code.

lehins commented on eras/shelley/impl/CHANGELOG.md:
```diff
@@ -2,6 +2,8 @@
 
 ## 1.17.0.0
 
+* Deprecate `Shelley.BlockChain` in favour of `Shelley.BlockBody`. #5156
```
```suggestion
* Deprecate `Cardano.Ledger.Shelley.BlockChain` in favour of `Cardano.Ledger.Shelley.BlockBody`. #5156
```

It is funny, my browser knows that I am in the US and suggests me that the British spelling of the word "favor" is wrong. :rofl: 
<img width="1180" height="196" alt="image" src="https://github.com/user-attachments/assets/a8371a8c-1960-4581-b9ee-d4ee143d5c7b" />

lehins commented on eras/allegra/impl/src/Cardano/Ledger/Allegra/BlockBody.hs:
```diff
@@ -0,0 +1,27 @@
+  ShelleyBlockBody (ShelleyBlockBody),
+  sbbHash,
+  sbbTxs,
+ )
```
Ditto

lehins commented on eras/babbage/impl/src/Cardano/Ledger/Babbage/Tx.hs:
```diff
@@ -63,11 +60,11 @@ instance AlonzoEraTx BabbageEra where
-  fromTxSeq = txSeqTxns
-  toTxSeq = AlonzoTxSeq
-  hashTxSeq = hashAlonzoTxSeq
+instance EraBlockBody BabbageEra where
```
Could you please move this instance into its own `BlockBody` module for the sake of consistency with other eras

lehins commented on eras/babbage/impl/src/Cardano/Ledger/Babbage/Tx.hs:
```diff
@@ -14,11 +14,8 @@ module Cardano.Ledger.Babbage.Tx (
 ) where
 
 import Cardano.Ledger.Allegra.Tx (validateTimelock)
+import Cardano.Ledger.Alonzo.BlockBody.Internal (AlonzoBlockBody (AlonzoBlockBody), abbHash, abbTxs)
```
Ditto. Importing safe variants only

lehins commented on eras/conway/impl/src/Cardano/Ledger/Conway/Tx.hs:
```diff
@@ -141,11 +138,11 @@ instance AlonzoEraTx ConwayEra where
   isValidTxL = conwayTxL . isValidAlonzoTxL
   {-# INLINE isValidTxL #-}
 
-instance EraSegWits ConwayEra where
```
Ditto. Move into its own module

lehins commented on eras/dijkstra/src/Cardano/Ledger/Dijkstra/Tx.hs:
```diff
@@ -71,11 +71,11 @@ instance AlonzoEraTx DijkstraEra where
   isValidTxL = dijkstraTxL . isValidAlonzoTxL
   {-# INLINE isValidTxL #-}
 
-instance EraSegWits DijkstraEra where
```
Ditto

lehins commented on eras/dijkstra/src/Cardano/Ledger/Dijkstra/Tx.hs:
```diff
@@ -9,6 +9,7 @@
 module Cardano.Ledger.Dijkstra.Tx (Tx (..)) where
 
 import Cardano.Ledger.Allegra.Tx (validateTimelock)
+import Cardano.Ledger.Alonzo.BlockBody.Internal (AlonzoBlockBody (AlonzoBlockBody), abbHash, abbTxs)
```
Ditto

lehins commented on eras/conway/impl/src/Cardano/Ledger/Conway/Tx.hs:
```diff
@@ -19,6 +19,7 @@ module Cardano.Ledger.Conway.Tx (
 ) where
 
 import Cardano.Ledger.Allegra.Tx (validateTimelock)
+import Cardano.Ledger.Alonzo.BlockBody.Internal (AlonzoBlockBody (AlonzoBlockBody), abbHash, abbTxs)
```
Ditto. No need to rely on internal module, export safe variants:
```suggestion
import Cardano.Ledger.Alonzo.BlockBody (
  AlonzoBlockBody (AlonzoBlockBody),
  alonzoBlockBodyHash,
  alonzoBlockBodyTxs
  )
```

lehins commented on eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxSeq.hs:
```diff
@@ -1,8 +1,32 @@
+txSeqTxns = (^. txSeqBlockBodyL)
+
+hashAlonzoTxSeq :: EraBlockBody era => BlockBody era -> Hash HASH EraIndependentBlockBody
+hashAlonzoTxSeq = hashBlockBody
```
Otherwise they are not equivalent
```suggestion
hashAlonzoTxSeq :: AlonzoBlockBody era -> Hash HASH EraIndependentBlockBody
hashAlonzoTxSeq = hashAlonzoBlockBody
```

lehins commented on eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxSeq.hs:
```diff
@@ -1,8 +1,32 @@
+pattern AlonzoTxSeq s = AlonzoBlockBody s
+
+txSeqTxns :: EraBlockBody era => BlockBody era -> StrictSeq (Tx era)
+txSeqTxns = (^. txSeqBlockBodyL)
```
```suggestion
txSeqTxns :: AlonzoBlockBody era -> StrictSeq (Tx era)
txSeqTxns = alonzoBlockBodyTxs
```

lehins commented on eras/shelley/impl/src/Cardano/Ledger/Shelley/BlockBody/Internal.hs:
This module was moved without preserving git history. This needs to be fixed. Let me know if you need help with this.

lehins reviewed (changes requested):
It is looking better.
However, there are still a few things that need to be improved

aniketd committed:
Move Shelley.BlockChain->Shelley.BlockBody.Internal

lehins reviewed (approved):
Perfect! Thank you!
